railties/test/application/configuration_test.rb

```diff
@@ -3819,6 +3826,74 @@ class Post < ActiveRecord::Base
       assert_not_includes ActiveRecord::Base.filter_attributes, :content
     end
 
+    test "ActiveRecord::Encryption.config is ready for encrypted attributes when app is lazy loaded" do
+      add_to_config <<-RUBY
+        config.enable_reloading = false
+        config.eager_load = false
+      RUBY
+
+      app_file "config/initializers/active_record.rb", <<-RUBY
+        Rails.application.config.active_record.encryption.primary_key = "dummy_key"
+        Rails.application.config.active_record.encryption.previous = [ { key_provider: MyOldKeyProvider.new } ]
+
+        ActiveRecord::Base.establish_connection(adapter: "sqlite3", database: ":memory:")
+        ActiveRecord::Migration.verbose = false
+        ActiveRecord::Schema.define(version: 1) do
+          create_table :posts do |t|
+            t.string :content
+          end
+        end
+
+        ActiveRecord::Base.connection.schema_cache.add("posts")
+      RUBY
+
+      app_file "app/models/post.rb", <<-RUBY
+        class Post < ActiveRecord::Base
+          encrypts :content, key_provider: MyCustomKeyProvider.new(ActiveRecord::Encryption.config.primary_key)
+        end
+      RUBY
+
+      app "development"
+
+      assert_kind_of ::MyOldKeyProvider, Post.attribute_types["content"].previous_schemes.first.key_provider
+      assert_kind_of ::MyCustomKeyProvider, Post.attribute_types["content"].scheme.key_provider
+      assert_equal "dummy_key", Post.attribute_types["content"].scheme.key_provider.primary_key
+    end
+
+    test "ActiveRecord::Encryption.config is ready for encrypted attributes when app is eager loaded" do
```

Comment:
This is the test which fails before the addition of the fix, with the following error:

```
ActiveRecord::Encryption::Errors::Configuration: Missing Active Record encryption credential: active_record_encryption.primary_key
    /workspaces/rails/activerecord/lib/active_record/encryption/config.rb:43:in `block (2 levels) in <class:Config>'
    /workspaces/rails/tmp/d20240106-41247-luvct9/app/app/models/post.rb:2:in `<class:Post>'
    /workspaces/rails/tmp/d20240106-41247-luvct9/app/app/models/post.rb:1:in `<top (required)>'
    /usr/local/rvm/gems/default/gems/zeitwerk-2.6.12/lib/zeitwerk/kernel.rb:30:in `require'
    /usr/local/rvm/gems/default/gems/zeitwerk-2.6.12/lib/zeitwerk/kernel.rb:30:in `require'
    /usr/local/rvm/gems/default/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/helpers.rb:135:in `const_get'
    /usr/local/rvm/gems/default/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/helpers.rb:135:in `cget'
    /usr/local/rvm/gems/default/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/eager_load.rb:175:in `block in actual_eager_load_dir'
    /usr/local/rvm/gems/default/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/helpers.rb:40:in `block in ls'
    /usr/local/rvm/gems/default/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/helpers.rb:25:in `each'
    /usr/local/rvm/gems/default/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/helpers.rb:25:in `ls'
    /usr/local/rvm/gems/default/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/eager_load.rb:170:in `actual_eager_load_dir'
    /usr/local/rvm/gems/default/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/eager_load.rb:17:in `block (2 levels) in eager_load'
    /usr/local/rvm/gems/default/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/eager_load.rb:16:in `each'
    /usr/local/rvm/gems/default/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/eager_load.rb:16:in `block in eager_load'
    /usr/local/rvm/gems/default/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/eager_load.rb:10:in `synchronize'
    /usr/local/rvm/gems/default/gems/zeitwerk-2.6.12/lib/zeitwerk/loader/eager_load.rb:10:in `eager_load'
    /usr/local/rvm/gems/default/gems/zeitwerk-2.6.12/lib/zeitwerk/loader.rb:379:in `block in eager_load_all'
    /usr/local/rvm/gems/default/gems/zeitwerk-2.6.12/lib/zeitwerk/loader.rb:377:in `each'
    /usr/local/rvm/gems/default/gems/zeitwerk-2.6.12/lib/zeitwerk/loader.rb:377:in `eager_load_all'
    /workspaces/rails/railties/lib/rails/application/finisher.rb:80:in `block in <module:Finisher>'
    /workspaces/rails/railties/lib/rails/initializable.rb:32:in `instance_exec'
    /workspaces/rails/railties/lib/rails/initializable.rb:32:in `run'
    /workspaces/rails/railties/lib/rails/initializable.rb:61:in `block in run_initializers'
    /usr/local/lib/ruby/3.2.0/tsort.rb:228:in `block in tsort_each'
    /usr/local/lib/ruby/3.2.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
    /usr/local/lib/ruby/3.2.0/tsort.rb:431:in `each_strongly_connected_component_from'
    /usr/local/lib/ruby/3.2.0/tsort.rb:349:in `block in each_strongly_connected_component'
    /usr/local/lib/ruby/3.2.0/tsort.rb:347:in `each'
    /usr/local/lib/ruby/3.2.0/tsort.rb:347:in `call'
    /usr/local/lib/ruby/3.2.0/tsort.rb:347:in `each_strongly_connected_component'
    /usr/local/lib/ruby/3.2.0/tsort.rb:226:in `tsort_each'
    /usr/local/lib/ruby/3.2.0/tsort.rb:205:in `tsort_each'
    /workspaces/rails/railties/lib/rails/initializable.rb:60:in `run_initializers'
    /workspaces/rails/railties/lib/rails/application.rb:423:in `initialize!'
    /workspaces/rails/tmp/d20240106-41247-luvct9/app/config/environment.rb:5:in `<top (required)>'
    /usr/local/rvm/gems/default/gems/zeitwerk-2.6.12/lib/zeitwerk/kernel.rb:38:in `require'
    /usr/local/rvm/gems/default/gems/zeitwerk-2.6.12/lib/zeitwerk/kernel.rb:38:in `require'
    /workspaces/rails/railties/test/application/configuration_test.rb:59:in `app'
    /workspaces/rails/railties/test/application/configuration_test.rb:3890:in `block in <class:ConfigurationTest>'
```